### PR TITLE
Refactor docs

### PR DESCRIPTION
**Pull request by jankishah-gds**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,131 +1,130 @@
 # The Relying Party Transmitter
 
-This is the source code for the Relying Party Transmitter (RPT) which is responsible for transmitting messages from Relying parties (RPs) to the Shared Signals Framework (SSF).
+The Relying Party Transmitter (RPT) application demonstrates how a [Security Event Token](https://datatracker.ietf.org/doc/html/rfc8417) (SET) can be sent from Relying Parties (RPs) to the Transmitter in the Shared Signals Framework (SSF) using a severless function, known as the Transmitter function. This function signs and encrypts a Security Event Token and forwards it to the Shared Signals Framework endpoint.
+
+Helper functions have been added for the purpose of testing the Transmitter function. The Generator function generates test cases of messages to pass on and the public key function simulates an RP's endpoint for serving their public key.
+
+Though this implementation uses serverless functions in AWS (AWS Lambda), the approach described is platform independent.
```
Comment (Cavalier3142) [inline README.md:7]: Platform-independent?

File: README.md [hunk 2]
```diff
@@ -1,131 +1,130 @@
...
+
 ![Architecture of the RPT](/READMEresources/MainArchitecture.png)
 
----
+## Background
 
-# 1. Setup
+A Security Event Token is issued on a state change of a security subject, for example a user account or an HTTP session. When the Shared Signals Transmitter endpoint receives a SET, it will validate and interpret the received Security Event Token and takes its own independent actions, if any.
```
Comment (Cavalier3142) [inline README.md:13]: SET? Abbreviation already "established" above.

Comment (Cavalier3142) [inline README.md:13]: Shared signals are --informational-- and not linked to actions. A transmitter and receiver can, together, agree an action on receipt of a particular message but this is --not-- part of the framework.

File: README.md [hunk 3]
```diff
@@ -1,131 +1,130 @@
...
 
----
+An example Security Event Token:
 
-## 1.1. Development Environment Set Up
+```json
+{
```
Comment (Cavalier3142) [inline README.md:18]: Might be worth explaining what each field in the set is.

File: README.md [hunk 4]
```diff
@@ -1,131 +1,130 @@
...
+   "SET": {
+      "iss": "https://issuer.digitalIdentity.gov/",
+      "jti": "756e697175654964656e74696669657230",
+      "iat": 1520364019,
+      "aud": "https://audience.hmrc.gov/",
+      "events": {
+      "https://schemas.digitalIdentity.gov/secevent/risc/event-type/ipv-spot-request-recieved": {
+         "subject": {
+            "format": "iss_sub",
```
Comment (Cavalier3142) [inline README.md:27]: The iss-sub user subject format has been deprecated in favour of { "format" : "uri", "uri" : "<some-user-uri>" }. Not entirely sure this is a user subject though? I might be inclined to use a different message type in the example here, maybe one that clearly has a user subject? Maybe RISC account disabled?

File: README.md [hunk 5]
```diff
@@ -1,131 +1,130 @@
...
 
-The following extensions should be installed for VSCode:
+The Transmitter function signs and encrypts a Security Event Token and forwards it to the Shared Signals Framework (SSF) Transmitter endpoint. It is triggered when a message is received from the Generator function.
 
-- [AWS Toolkit](https://docs.aws.amazon.com/toolkit-for-vscode/latest/userguide/welcome.html) - The AWS Toolkit can be connected through SSO using `aws configure sso` and then selecting the correct profile at the bottom of VSCode.
+This function:
 
-- [SonarLint](https://marketplace.visualstudio.com/items?itemName=SonarSource.sonarlint-vscode) - SonarLint should be connected through the `Connected Mode` it offers. Once it has been connected, creating the binding of your local repo to the `di-fraud-mock-rp` repo on `SonarCloud`.
-  This should add `problems`, which is accessed by `View` -> `Problems`.
+1. Signs the Security Event Token, using a private key as the signature. The signed SET and signature are used to generate a JWS object. The SSF endpoint will use the corresponding public key to verify the signature.
```
Comment (Cavalier3142) [inline README.md:52]: The private key is used to produce the signature. The private key is known to the RP only and will definitely not be included in the JWS! Of course in the RPT tool we need access to the private key to sign.

File: README.md [hunk 6]
```diff
@@ -1,131 +1,130 @@
...
+2. Encrypts the JWS, using the full JWS object as the message:
+   1. Generates a random Content Encryption Key (CEK).
+   2. Encrypt the CEK with the RP's public key to form the JWE encrypted key.
+   3. Generates a random JWE inititilization vector.
```
Comment (Cavalier3142) [inline README.md:56]: Spelling.

File: README.md [hunk 7]
```diff
@@ -1,131 +1,130 @@
...
 
-- [AWS Toolkit](https://docs.aws.amazon.com/toolkit-for-vscode/latest/userguide/welcome.html) - The AWS Toolkit can be connected through SSO using `aws configure sso` and then selecting the correct profile at the bottom of VSCode.
+This function:
 
-- [SonarLint](https://marketplace.visualstudio.com/items?itemName=SonarSource.sonarlint-vscode) - SonarLint should be connected through the `Connected Mode` it offers. Once it has been connected, creating the binding of your local repo to the `di-fraud-mock-rp` repo on `SonarCloud`.
-  This should add `problems`, which is accessed by `View` -> `Problems`.
+1. Signs the Security Event Token, using a private key as the signature. The signed SET and signature are used to generate a JWS object. The SSF endpoint will use the corresponding public key to verify the signature.
+2. Encrypts the JWS, using the full JWS object as the message:
+   1. Generates a random Content Encryption Key (CEK).
+   2. Encrypt the CEK with the RP's public key to form the JWE encrypted key.
```
Comment (Cavalier3142) [inline README.md:55]: The payload is signed and not encrypted?

File: README.md [hunk 8]
```diff
@@ -1,131 +1,130 @@
...
+3. Sends the JWE to the SSF endpoint.
 
-## 1.2. Local Build and Invocation
+## Helper functions
 
-### Pre-Commit
+### Generator function
 
-Install `pre-commit` to local repo at root directory
+The Generator function generates a series of messages to sent to the Transmitter function. It is triggered by an API call.
```
Comment (Cavalier3142) [inline README.md:67]: to sent -> to send. Should Transmitter be capitalised?

File: README.md [hunk 9]
```diff
@@ -339,57 +220,60 @@ Capturing Lambda handlers can be done by adding the captureLambdaHandler middlew
...
-> ✅ `let a = await Promise.resolve(func())`
+### Configure recommended developer tools
 
-### Use `console.log` to ensure `type` is correct.
+- Run `pre-commit install` to install pre-commit hooks
+- Install the [AWS Toolkit extension for VS Code](https://docs.aws.amazon.com/toolkit-for-vscode/latest/userguide/welcome.html)
+- The AWS Toolkit can be connected through SSO using `aws configure sso` and then selecting the correct profile at the bottom of VSCode.
+- Install the [SonarLint extension for VS Code](https://marketplace.visualstudio.com/items?itemName=SonarSource.sonarlint-vscode)
+- SonarLint should be connected through the `Connected Mode` it offers.
+- Once it has been connected, creating the binding of your local repo to the di-fraud-mock-rp repo on SonarCloud
```
Comment (Cavalier3142) [inline README.md:256]: creating -> create?

**Review by Cavalier3142 — APPROVED** [review]
Looks good! Have added a few picky comments for your consideration.

File: README.md (hunk 7 above)
Comment (jankishah-gds) [reply README.md]: I've spoken with Dan Lewis to get more clarification around what the transmitter function is doing. He confirmed that we used to do encryption but no longer do and also gave information about how cognito is used to generate an auth key. I've updated the docs to reflect this.

**Review by Cavalier3142 — APPROVED** [review]
👍 looks good.

**Review by Cavalier3142 — APPROVED** [review]
👍

**Pull request merged**